boxoffice/views/order.py

```diff
@@ -122,6 +125,11 @@ def order(item_collection):
     if not buyer_form.validate():
         return make_response(jsonify(message='Invalid buyer details'), 400)
 
+    for line_item_form in line_item_forms:
+        item = Item.query.get(line_item_form.data.get('item_id'))
+        if (item.get_confirmed_line_items.count() + line_item_form.data.get('quantity')) > item.quantity_total:
+            return make_response(jsonify(error_type='order_calculation', message='The selected quantity for {item} is not available. Please edit the order and update the quantity.'.format(item=item.title)), 400)
```

Comment:
This block is very db-inefficient. It makes two separate database calls for every line item. This could be entirely one call.